dsp/modules/gpt3.py

```diff
@@ -157,7 +157,7 @@ def __call__(
         if only_completed and len(completed_choices):
             choices = completed_choices
 
-        completions = [self._get_choice_text(c) for c in choices]
+        completions = [self.get_choice_text(c) for c in choices]
```

Comment:
This kind of changes the behavior right? It used to be "stripped", now it has space before it.

I'm not sure if that will affect things in unexpected ways. IMO this whole refactor is great but shouldn't be part of this PR, it can be another PR though.

Reply:
you are right, now I see that it is not equivalent, I will revert the refactor, and leave only the llama_cpp implementation.